hooks/useTests.ts

```diff
@@ -0,0 +1,38 @@
+import { graphqlRequest, clientGraphqlRequest } from '../lib/harsura'
+import { gql } from 'graphql-request'
+import { useQuery } from '@tanstack/react-query'
+
+export const DEFAULT_TESTS_LIMIT = 3
+export interface Test {
+  id: number
+  name: string
+}
+
+const testsQueryDocument = gql`
+  query getTests($limit: Int = 10) {
+    tests(limit: $limit) {
+      id
+      name
+    }
+  }
+`
+
+const fetchTests = async (limit = DEFAULT_TESTS_LIMIT) => {
```

Comment:
`await `is ignored in this function, actually we don't need define `async `here, please remove